File: examples/governance/operations_domain/lib/operations_domain/incident/commands/report_incident.rb
module OperationsDomain
  class Incident
    module Commands
      class ReportIncident
        include Hecks::Command
        emits "ReportedIncident"

        attr_reader :model_id
        attr_reader :severity
        attr_reader :category
        attr_reader :description
        attr_reader :reported_by_id

        def initialize(
          model_id: nil,
          severity: nil,
          category: nil,
          description: nil,
          reported_by_id: nil
        )
          @model_id = model_id
          @severity = severity
          @category = category
          @description = description
          @reported_by_id = reported_by_id
        end

        def call
          Incident.new(
            model_id: model_id,
            severity: severity,
            category: category,
            description: description,
            reported_by_id: reported_by_id,
            reported_at: Time.now.to_s,
            status: "reported"
          )
        end
      end
    end
  end
end

File: examples/governance_static_go/domain/report_incident.go
package domain

import (
	"time"
)

type ReportIncident struct {
	ModelId string `json:"model_id"`
	Severity string `json:"severity"`
	Category string `json:"category"`
	Description string `json:"description"`
	ReportedById string `json:"reported_by_id"`
}

func (c ReportIncident) CommandName() string { return "ReportIncident" }

func (c ReportIncident) Execute(repo IncidentRepository) (*Incident, *ReportedIncident, error) {
	agg := NewIncident(c.ModelId, c.Severity, c.Category, c.Description, c.ReportedById, time.Time{}, time.Time{}, "", "", "")
	if err := agg.Validate(); err != nil {
		return nil, nil, err
	}
	if err := repo.Save(agg); err != nil {
		return nil, nil, err
	}
	event := ReportedIncident{
		AggregateID: agg.ID,
		ModelId: c.ModelId,
		Severity: c.Severity,
		Category: c.Category,
		Description: c.Description,
		ReportedById: c.ReportedById,
		OccurredAt: time.Now(),
	}
	return agg, &event, nil
}
